funmi5/ncc-media-team
# #169202526 #170071446 Build out endpoints for reset/forgot password and user profiles

**What does this PR do?**
This PR builds out the endpoints for reset password and profiles

**Description of tasks**
- build endpoints for forgot password and reset password
- build a function to integrate automated sendgrid mails
- build out CRUD endpoints for user profiles
- add a circle ci config file
- add a model relationship for users - profiles
- add more authentication functions to validate the endpoints

**How should this PR be tested/checked?**
Clone the project and run npm install
Run npm run start:dev
Make sure to update your .env file
To run the migration, run npm run migrate
To run the seeders, run npm run seed
To test: npm run test

**NOTE:**
- Make sure to include your signed in token in the `Bearer token`
Go to postman and run this endpoint: POST `http://localhost:3000/api/v1/users/forgotpassword`
using this sample resource:

```
{
email: "<the email representing the password account you wish to reset>"
}
```
with the url generated, make a POST request with it
using this sample resource :-
```
{
	"password": "Funmi.Bl2",
	"confirmPassword": "Funmi.Bl2"
}

```

**For profiles**
to create a profile:
POST `localhost:3000/api/v1/profile/<user-id>`
```
{
   "picture": "http://shttegggggge",
   "phoneNumber": "08000000000",
   "sideNote": "At the darkest hour, light will shine!",
   "dateOfBirth": "2019-09-14T00:43:23.827Z"
}
```
to update a profile:
PATCH `localhost:3000/api/v1/profile/<user-id>`
```
{
   "picture": "http://shttegggggge",
   "phoneNumber": "2222222222",
   "sideNote": "At the darkest hour, light will shine!",
   "dateOfBirth": "2019-09-14T00:43:23.827Z"
}
```
to get the profile: GET `localhost:3000/api/v1/profile/<user-id>`
to get all profiles: GET `localhost:3000/api/v1/profiles`


**What are the relevant pivotal tracker stories?**
<a href="https://www.pivotaltracker.com/story/show/169202526">169202526</a>
<a href = "https://www.pivotaltracker.com/story/show/170071446">170071446</a>
